src/classes/HouseholdService.cls

```diff
@@ -0,0 +1,266 @@
+        List<Contact> contactsToBeUpdatedWithHouseholdId = new List<Contact>();
+        Integer i = 0;
+        for (Contact con : getContactsThatAreNotExcludedByRecordType(contacts)) {
+            Database.SaveResult saveResult = householdSaveResults[i];
+            if (saveResult.isSuccess() == true) {
+                //write the new Household Ids to the Contacts
+                Contact cloneContact = con.Clone(true, false);
+                cloneContact.npo02__household__c = saveResult.getId();
+                contactsToBeUpdatedWithHouseholdId.add(cloneContact);
+            } else {
+                Database.Error err = saveResult.getErrors()[0];
+                con.addError(err.getMessage());
+            }
+            i += 1;
+        }
+        return contactsToBeUpdatedWithHouseholdId;
+    }
+
+    private List<npo02__Household__c> getHouseholdsForInsert(List<Contact> contacts) {
+        List<npo02__household__c> households = new List<npo02__household__c>();
+        for (Contact con : getContactsThatAreNotExcludedByRecordType(contacts)) {
+            households.add(getHouseholdForInsertFromContact(con));
+        }
+        return households;
+    }
+
+    private List<Contact> getContactsThatAreNotExcludedByRecordType(List<Contact> contacts) {
+        List<Contact> contactsThatAreNotExcludedByRecordType = new List<Contact>();
+        for (Contact con : contacts) {
+            if (isNotExcludedByRecordType(con)) {
+                contactsThatAreNotExcludedByRecordType.add(con);
+            }
+        }
+        return contactsThatAreNotExcludedByRecordType;
+    }
+
+    private npo02__Household__c getHouseholdForInsertFromContact(Contact con) {
+        npo02__household__c household =
+                new npo02__household__c(Name = getDefaultHouseholdName(con));
+        syncPrimaryAddressBlock(household, con);
+        return household;
+    }
+
+    private void syncPrimaryAddressBlock(npo02__Household__c household, Contact con) {
+        household.npo02__MailingStreet__c = con.MailingStreet;
+        household.npo02__MailingCity__c = con.MailingCity;
+        household.npo02__MailingState__c = con.MailingState;
+        household.npo02__MailingPostalCode__c = con.MailingPostalCode;
+        household.npo02__MailingCountry__c = con.MailingCountry;
+        household.npo02__HouseholdPhone__c = con.HomePhone;
+        household.npo02__HouseholdEmail__c = con.Email;
+    }
+
+    private String getDefaultHouseholdName(Contact con) {
+        String hName = con.LastName;
+        return hName += ' ' + System.Label.npo02.DefaultHouseholdName;
+    }
+
+    private Boolean isNotExcludedByRecordType(Contact con) {
+        return !isRecordTypesOnContacts()
+                || !settings.contactRecordTypesToExclude().contains(recordTypeIdFor(con));
+    }
+
+    private Id recordTypeIdFor(Contact con) {
+        return (Id) con.get(RECORD_TYPE_ID);
+    }
+
+    private Boolean isRecordTypesOnContacts() {
```

Comment:
@kenlewis-sfdo Calling getMap() more than once is going to be a cpu expensive operation. I recommend an alternative approach using a static var so that the evaluation of whether the field exists only happens once. You could also compare the cpu time of calls to getMap() vs. just wrapping a call to `get('RecordTypeId')` in a try/catch block.